### PR TITLE
Automated cherry pick of #2403: Fix: incompatible job spec caused controller to fail

### PR DESCRIPTION
**Pull request by rene-dekker**
Cherry pick of #2403 on release-v1.28.

#2403: Fix: incompatible job spec caused controller to fail

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Review by tmjd — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
